src/api/libcellml/importsource.h

```diff
@@ -109,6 +117,161 @@ class LIBCELLML_EXPORT ImportSource: public Entity
      */
     ImportSourcePtr clone() const;
 
+    /**
+     * @brief Add a component to this import source item.
+     *
+     * The added @p component will be treated as an import, and
+     * will be sourced from the combination of url of this import source
+     * item, and the component's own import reference.
+     *
+     * @param component The @c ComponentPtr to add.
+     *
+     * @return @c true if the component was added successfully, @c false
+     * if it already exists in this import.
+     */
+    bool addComponent(const ComponentPtr &component);
+
+    /**
+     * @brief Remove the component at the given @p index.
+     *
+     * Remove the component from this import source at the given @p index.
+     * @p index must be in the range [0, \#component).
+     *
+     * @param index The index of the component to remove.
+     * @param setEmpty Whether or not to set the removed Component's
+     *        importSource to nullptr.  True by default.
+     *
+     * @return True if the component was removed, false otherwise.
+     */
+    bool removeComponent(size_t index, bool setEmpty = true);
+
+    /**
+     * @overload
+     *
+     * @brief Remove the component with the given pointer.
+     *
+     * Remove the component with the pointer @p component.
+     *
+     * @param component The pointer to the component to remove.
+     * @param setEmpty Whether or not to set the removed Component's
+     *        importSource to nullptr.  True by default.
+     *
+     * @return True if the component was removed, false otherwise.
+     */
+    bool removeComponent(ComponentPtr &component, bool setEmpty = true);
+
+    /**
+     * @brief Remove all components from the import source.
+     *
+     * Note that components which are removed from the import will have their
+     * import source cleared.  They thus become locally
+     * concrete instances in the model, and will return @c false to the
+     * function isImport().
```

Comment:
Need to properly reference `isImport()`.

Reply:
I don't understand?

Reply:
Sorry, I meant something like `sa isImport`?